Add enovy ai gateway (helm dependencies + doc) (#360)
InftyAI/llmaz: docs/envoy-ai-gateway.md

# Envoy AI Gateway

[Envoy AI Gateway](https://aigateway.envoyproxy.io/) is an open source project for using Envoy Gateway
to handle request traffic from application clients to Generative AI services.

> ## Prerequisites

Reviewer: Remove this since no requirements here.

## How to use

### 1. Enable Envoy Gateway and Envoy AI Gateway in llmaz Helm

> Enable Envoy Gateway and Envoy AI Gateway in the `values.global.yaml` file, envoy gateway and envoy ai gateway are disabled by default.

Reviewer: enabled by default.

> Enable Envoy Gateway and Envoy AI Gateway in the `values.global.yaml` file, envoy gateway and envoy ai gateway are disabled by default.

Reviewer 2: maybe we can add link to `values.global.yaml`?

>     ```yaml

Reviewer: Fix the indent here.

    envoy-gateway:
      enabled: true
    envoy-ai-gateway:
      enabled: true
    ```

>     Enovy Gateway installation with helm can be done standalone by https://gateway.envoyproxy.io/latest/install/install-helm/. Envoy AI Gateway follow https://aigateway.envoyproxy.io/docs/getting-started/.

Reviewer: ```suggestion
Note: [Envoy Gateway installation](https://gateway.envoyproxy.io/latest/install/install-helm/) and [Envoy AI Gateway installation](https://aigateway.envoyproxy.io/docs/getting-started/) can be done standalone.
```

> 

Reviewer 2: <img width="1063" alt="image" src="https://github.com/user-attachments/assets/be5787c9-81c9-498c-be0b-3af897fb7706" />

Is this a formatting issue or is it expected to be displayed this way? 🤔

### 2. Check Envoy Gateway and Envoy AI Gateway

>  Run `kubectl wait --timeout=5m -n envoy-gateway-system deployment/envoy-gateway --for=condition=Available` to wait for the envoy gateway to be ready.

Reviewer: Miss the check for envoy ai gateway.

Install the GatewayClass, Gateway, HTTPRoute and example app:
```bash

> kubectl apply -f https://github.com/envoyproxy/gateway/releases/download/latest/quickstart.yaml -n default

Reviewer: Let's remove this, we only focus on llmaz project.

```

### 3. Basic AI Gateway example

To expose your model(Playground) to Envoy Gateway, you need to create a GatewayClass, Gateway, and AIGatewayRoute. The following example shows how to do this.

Example configuration for a basic AI Gateway:
 - model: `qwen2-0.5b`; backend ref name is also `qwen2-0--5b`
 - model lb service: `qwen2-0--5b-lb`

```bash

> cat <<EOF | kubectl apply -f -

Reviewer: Move all the related files to the `docs/examples/envoy-ai-gateway`, they can be placed in one file. And we still miss the definition for the models/playgrounds.
